pkg/operator/controllers/checkers/ingresscertificatechecker/checker.go

```diff
@@ -59,13 +62,41 @@ func (r *checker) Check(ctx context.Context) error {
 		return err
 	}
 
-	if ingress.Spec.DefaultCertificate == nil {
+	cluster, err := r.arocli.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	return validateCertificate(cv.Spec.ClusterID, ingress.Spec.DefaultCertificate, cluster.Spec.Domain)
+}
+
+func validateCertificate(clusterId configv1.ClusterID, certificate *corev1.LocalObjectReference, clusterDomain string) error {
+	clusterHasManagedDomain := clusterHasManagedDomain(clusterDomain)
+
+	if certificate == nil {
+		if clusterHasManagedDomain {
+			return fmt.Errorf("ARO has not provisioned the default ingress certificate because it is empty and cluster has a managed domain")
+		}
 		return errNoDefaultCertificate
 	}
 
-	if ingress.Spec.DefaultCertificate.Name != string(cv.Spec.ClusterID)+"-ingress" {
-		return fmt.Errorf("custom ingress certificate in use: %q", ingress.Spec.DefaultCertificate.Name)
+	certificateName := certificate.Name
+	certificateNameIsValid := certificateNameIsValid(certificateName, clusterId)
+
+	if clusterHasManagedDomain && !certificateNameIsValid {
+		return fmt.Errorf("custom ingress certificate in use: %q", certificateName)
 	}
 
 	return nil
 }
+
+func certificateNameIsValid(certificateName string, clusterID configv1.ClusterID) bool {
+	ingressNameSuffix := "-ingress"
+	expectedCertName := string(clusterID) + ingressNameSuffix
```

Comment:
that's a little too much here `string(clusterID) + "-ingress"` would be sufficient

Reply:
Declaring ingressNameSuffix as a package constant as I don't like magic hardcoded strings.
Tks!